OpenRA.Game/Traits/CreatesShroud.cs

```diff
@@ -15,22 +15,30 @@ namespace OpenRA.Traits
 	public class CreatesShroudInfo : ITraitInfo
 	{
 		public readonly WRange Range = WRange.Zero;
-		public object Create(ActorInitializer init) { return new CreatesShroud(this); }
+
+		public object Create(ActorInitializer init) { return new CreatesShroud(init.Self, this); }
 	}
 
 	public class CreatesShroud : ITick, ISync
 	{
-		CreatesShroudInfo info;
+		readonly CreatesShroudInfo info;
+		readonly bool lobbyShroud;
+		readonly bool lobbyFog;
 		[Sync] CPos cachedLocation;
 		[Sync] bool cachedDisabled;
 
-		public CreatesShroud(CreatesShroudInfo info)
+		public CreatesShroud(Actor self, CreatesShroudInfo info)
 		{
 			this.info = info;
+			lobbyShroud = self.World.LobbyInfo.GlobalSettings.Shroud;
+			lobbyFog = self.World.LobbyInfo.GlobalSettings.Fog;
 		}
 
 		public void Tick(Actor self)
 		{
+			if (!lobbyShroud && !lobbyFog)
```

Comment:
May as well do this logic in the constructor and have a single boolean field for the shroud and fog being disabled.